pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -168,6 +168,10 @@ func (r *Reconciler) create() error {
 }
 
 func (r *Reconciler) update() error {
+	if err := validateMachine(*r.machine, *r.providerSpec); err != nil {
```

Comment:
what's the rationale behind introducing this in the PR?

Reply:
We have this validation in other providers, and tests for actuator events cover this case. I'd like to keep these tests consistent among all providers.